docs/decisions/0006-middleware-support-for-http-clients.rst

```diff
@@ -0,0 +1,41 @@
+Middleware Support for HTTP client
+==========================================
+
+Status
+------
+
+Proposed
+
+Context
+-------
+
+We currently expose a HTTP client(axios instance) via ``getAuthenticatedHttpClient`` and ``getHttpClient`` used in
+our MFEs to make HTTP requests. Since Open edX uses Django for its backend services, the MFEs have to handle snake-cased <-> camelCase
+conversions manually. The `axios-case-converter <https://www.npmjs.com/package/axios-case-converter>` package provides a middleware that would handle the conversion automatically. In the
+process of integrating the package, we thought it might be beneficial to
+establish a pattern for supporting middleware.
+
+Decision
+--------
+
+``getAuthenticatedHttpClient`` and ``getHttpClient`` take an ``options`` argument that includes configuration for the client.
+Only ``useCache`` is currently supported. We will introduce a new option ``middleware``, which will be a list of middleware
+that will be applied to the base axios instance. Each item in the list can either be a function or an array in the form of [function, options].
+We will apply the middleware and handle any exceptions that occur.
+
+It will look like this::
+
+    {
```

Comment:
One other thought for this code block is to put in the context of calling `getAuthenticatedHttpClient` as a consumer, so the depicted code isn't just the options object in isolation.

```jsx
const httpClient = getAuthenticatedHttpClient({
  useCache: true,
  middleware: [
    axiosConvertCase,
    [axiosRetry, { retries: 3 }],
    // ...
  ],
});